### PR TITLE
[usm] extract http2 protocol

### PR DESCRIPTION
**Pull request by guyarb**
### What does this PR do?

Implements USM's http2 support as a new program inside USM.

### Motivation

We're working on a larger refactor to improve readability of USM, along with allowing enabling features easily.
Now we're addressing http2 support.

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature. 
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

File: dummy [hunk 1]
```diff
@@ -0,0 +1 @@
+a
```
Comment (DanielLavie) [inline dummy:1]: Remove

File: pkg/network/usm/monitor.go [hunk 2]
```diff
@@ -10,20 +10,16 @@ package usm
 import (
 	"errors"
 	"fmt"
-	"syscall"
-	"time"
-	"unsafe"
-
 	"github.com/cilium/ebpf"
 	"go.uber.org/atomic"
+	"syscall"
```
Comment (DanielLavie) [inline pkg/network/usm/monitor.go:15]: Fix order

**Review by DanielLavie — CHANGES_REQUESTED** [review]

**Review by hmahmood — APPROVED** [review]

**Review by derekwbrown — APPROVED** [review]

File: dummy (hunk 1 above)
Comment (guyarb) [reply dummy]: done

File: pkg/network/usm/monitor.go (hunk 2 above)
Comment (guyarb) [reply pkg/network/usm/monitor.go:15]: done

File: pkg/network/protocols/http2/protocol.go [hunk 3]
```diff
@@ -0,0 +1,272 @@
...
+
+	return &protocol{
+		cfg:       cfg,
+		telemetry: telemetry,
+	}, nil
+}
+
+// ConfigureOptions add the necessary options for the http monitoring to work,
+// to be used by the manager. These are:
+// - Set the `http_in_flight` map size to the value of the `max_tracked_connection` configuration variable.
```
[the remark this answers is not in the document]
Comment (guyarb) [reply pkg/network/protocols/http2/protocol.go:89]: done

File: pkg/network/protocols/grpc/monitor_test.go [hunk 4]
```diff
@@ -10,6 +10,7 @@ package grpc
 import (
 	"context"
 	"fmt"
+	"github.com/DataDog/datadog-agent/pkg/network/protocols"
```
[the remark this answers is not in the document]
Comment (guyarb) [reply pkg/network/protocols/grpc/monitor_test.go:13]: done

**Review by DanielLavie — APPROVED** [review]

**Pull request merged**